test/AspNet.Security.OAuth.Providers.Tests/Douyin/DouyinTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace AspNet.Security.OAuth.Douyin;
+
+public class DouyinTests(ITestOutputHelper outputHelper) : OAuthTests<DouyinAuthenticationOptions>(outputHelper)
```

Comment:
For providers that override `BuildChallengeUrl()`, we also add tests for that.

For example:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/e6c467f08d52678a63dd843c8ae06b6a57d32629/test/AspNet.Security.OAuth.Providers.Tests/Xero/XeroTests.cs#L46